File: Leetcode_Medium/_128. Longest Consecutive Sequence/Solution.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num_set = set(nums)
        max_length = 0
    
        # Iterate through the array
        for num in nums:
            if num - 1 not in num_set:  # Check if the current number is the start of a sequence
                current_num = num
                current_length = 1
            
                # Count consecutive elements
                while current_num + 1 in num_set:
                    current_num += 1
                    current_length += 1
                
                max_length = max(max_length, current_length)  # Update max length
    
        return max_length
        

File: Leetcode_Medium/_128. Longest Consecutive Sequence/Solution.java
import java.util.HashSet;

public class Solution{

    public static int longestConsecutive(int[] nums) {
        HashSet<Integer> numSet = new HashSet<>();
        int maxLength = 0;

        // Add all numbers to the set
        for (int num : nums) {
            numSet.add(num);
        }

        // Iterate through the array
        for (int num : nums) {
            if (!numSet.contains(num - 1)) { // Check if the current number is the start of a sequence
                int currentNum = num;
                int currentLength = 1;

                // Count consecutive elements
                while (numSet.contains(currentNum + 1)) {
                    currentNum++;
                    currentLength++;
                }

                maxLength = Math.max(maxLength, currentLength); // Update max length
            }
        }

        return maxLength;
    }
}
